src/types/iterator.rs

```diff
@@ -213,4 +229,30 @@ mod tests {
         let iter: &PyIterator = PyIterator::try_from(obj.as_ref(py)).unwrap();
         assert_eq!(obj, iter.into());
     }
+
+    #[test]
+    fn test_as_ref() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
```

Comment:
We prefer `Python::with_gil` in new tests (`acquire_gil` is probably going to be deprecated or marked `unsafe` soon).

```suggestion
        Python::with_gil(|py| {
```